os/arch/arm/src/armv8-m/up_assert.c

```diff
@@ -160,6 +160,9 @@ static void up_stackdump(uint32_t sp, uint32_t stack_base)
 	uint32_t stack;
 
 	for (stack = sp & ~0x1f; stack < stack_base; stack += 32) {
+		while (stack < sp) {
+			stack += 4;
+		}
```

Comment:
This looks something strange.
The for loop sets initial value with `& ~0x1f` to align. But this newly added code ignores that. Then why does `& ~0x1f`?
Please confirm previous values of the sp are unnecessary first.